Code/Framework/AzCore/AzCore/Component/Component.h

```diff
@@ -118,6 +118,13 @@ namespace AZ
          */
         void SetId(const ComponentId& id)   { m_id = id; }
 
+        virtual void SetSerializedIdentifier(const AZStd::string&){};
+
+        virtual AZStd::string GetSerializedIdentifier()
+        {
+            return AZStd::string();
+        }
```

Comment:
To help with build times we should move implementations to cpp files whenever possible.